insights/parsers/engine_config.py

```diff
@@ -0,0 +1,187 @@
+            try:
+                key, val, ver = itemgetter(0, 1, -1)(line.split(' '))
+                self.fields.append(self.keyvalue(key.strip(':'), val, ver, key.strip(':').lower()))  # noqa
+            except:
+                pass
+        self.keywords = set([kw.kw_lower for kw in self.fields])
+
+    def __contains__(self, keyword):
+        return keyword.lower() in self.keywords
+
+    def __iter__(self):
+        return iter(self.fields)
+
+    def __getitem__(self, keyword):
+        kw = keyword.lower()
+        if kw in self.keywords:
+            return filter(
+                None,
+                [kv.value for kv in self.fields if kv.kw_lower == kw])
+
+    def head(self, keyword):
+        """Get first element from values(list)
+
+        Example::
+
+            >>> output['HotUnplugCpuSupported']
+            ['{"x86":"false","ppc":"false"}', '{"x86":"false","ppc":"false"}', '{"x86":"true","ppc":"true"}']
+            >>> output.head('HotUnplugCpuSupported')
+            '{"x86":"false","ppc":"false"}'
+
+        Args:
+
+            keyword (str): A key. Ex: 'HotUnplugCpuSupported'.
+
+        Returns:
+
+            str: First element from values(list) associated with a key.
+        """
+        values = self.__getitem__(keyword)
+        if values:
+            return values[0]
+
+    def last(self, keyword):
+        """Get last element from values(list)
+
+        Example::
+
+            >>> output['HotUnplugCpuSupported']
+            ['{"x86":"false","ppc":"false"}', '{"x86":"false","ppc":"false"}', '{"x86":"true","ppc":"true"}']
+            >>> output.last('HotUnplugCpuSupported')
+            '{"x86":"true","ppc":"true"}'
+
+        Args:
+
+            keyword (str): A key. Ex: 'HotUnplugCpuSupported'.
+
+        Returns:
+
+            str: Last element from values(list) associated with a key.
+        """
+        values = self.__getitem__(keyword)
+        if values:
+            return values[-1]
+
+    def get_version(self, keyword):
+        """Get versions associated with a key.
+
+        Typical output is ``engine-config --all`` command::
+
+            MaxStorageVdsTimeoutCheckSec: 30 version: general
+            HotUnplugCpuSupported: {"x86":"false","ppc":"false"} version: 3.6
+            HotUnplugCpuSupported: {"x86":"false","ppc":"false"} version: 4.0
+            HotUnplugCpuSupported: {"x86":"true","ppc":"true"} version: 4.1
+
+        Examples::
+
+            >>> output.get_version('MaxStorageVdsTimeoutCheckSec')
+            ['general']
+            >>> output.get_version('HotUnplugCpuSupported')
+            ['3.6', '4.0', '4.1']
+
+        Args:
+
+            keyword (str): A key. Ex: 'HotUnplugCpuSupported'.
+
+        Returns:
+
+            list: List of versions associated with a key.
+        """
+        kw = keyword.lower()
+        if kw in self.keywords:
+            return filter(
+                None,
+                [kv.version for kv in self.fields if kv.kw_lower == kw])
```

Comment:
Same questions as above regarding the default return value and use of the filter.